src/CoreWCF.UnixDomainSocket/src/CoreWCF/UnixDomainSocketBinding.cs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+using System.Xml;
+using CoreWCF.Channels;
+
+namespace CoreWCF
+{
+    public class UnixDomainSocketBinding : Binding
+    {
+        // private BindingElements
+        private UnixDomainSocketTransportBindingElement _transport;
+        private BinaryMessageEncodingBindingElement _encoding;
+        private UnixDomainSocketSecurity _security = new UnixDomainSocketSecurity();
+
+        public UnixDomainSocketBinding() { Initialize(); }
+        public UnixDomainSocketBinding(SecurityMode securityMode)
+            : this()
+        {
+            _security.Mode = securityMode;
+        }
+
+        public TransferMode TransferMode
+        {
+            get { return _transport.TransferMode; }
+            set { _transport.TransferMode = value; }
+        }
+
+        public HostNameComparisonMode HostNameComparisonMode
+        {
+            get { return _transport.HostNameComparisonMode; }
+            set { _transport.HostNameComparisonMode = value; }
+        }
+
+        [DefaultValue(TransportDefaults.MaxBufferPoolSize)]
+        public long MaxBufferPoolSize
+        {
```

Comment:
I think we can remove MaxBufferPoolSize, MaxBufferSize and MaxConnections. If you look at the named pipe code, some of these moved to the UseNetNamedPipes configuration options. But there we own the transport. Anything set there needs to be modifiable on the socket.